commands/scanpullrequest_test.go

```diff
@@ -82,6 +85,74 @@ func TestGetNewViolations(t *testing.T) {
 	assert.ElementsMatch(t, []string{"component-C", "component-D"}, []string{impactedPackageOne, impactedPackageTwo})
 }
 
+func TestGetNewViolationsCaseNoPrevViolations(t *testing.T) {
+	// Previous scan with no violation
+	previousScan := services.ScanResponse{
+		Violations: []services.Violation{},
+	}
+
+	// Current scan with 2 violations - XRAY-1 and XRAY-2
+	currentScan := services.ScanResponse{
+		Violations: []services.Violation{
+			{
+				IssueId:       "XRAY-1",
+				Summary:       "summary-1",
+				Severity:      "high",
+				ViolationType: "security",
+				Components:    map[string]services.Component{"component-A": {}},
+			},
+			{
+				IssueId:       "XRAY-2",
+				Summary:       "summary-2",
+				ViolationType: "security",
+				Severity:      "low",
+				Components:    map[string]services.Component{"component-C": {}},
+			},
+		},
+	}
+
+	// Run getNewViolations and expect both XRAY-1 and XRAY-2 violation in the results
+	newViolations := getNewViolations(previousScan, currentScan)
+	assert.Len(t, newViolations, 2)
+	assert.Equal(t, "XRAY-1", newViolations[0].IssueId)
```

Comment:
Are you sure that Xray-1 is always in location 0?
In TestGetNewViolations, I used ElementsMatch because of the inconsistency: https://github.com/jfrog/frogbot/blob/v1.0.0-alpha.2/commands/scanpullrequest_test.go#L82

Reply:
The table should be sorted, I expect that the higher severity will always be displayed first. 
Seems like the code should be fixed and not the test. @yahavi what do you think?

Reply:
You are right. In my case, the severity was the same for both issues. Your case is different. Let's keep it as is.